tuf/ngclient/updater.py

```diff
@@ -471,7 +503,67 @@ def _preorder_depth_first_walk(
         # If this point is reached then target is not found, return None
         return None
 
+    def _download_directory(self, folder: str) -> None:
+        # Downloading all metadata inside the specified folder
+
+        path = (
+            f"{BASE_URL}/repository_data/TAP 14"  # File path to download from
+        )
+        path = os.path.join(path, folder)  # Add the folder to the path
+
+        # Repository might be storing data in different formats.
+
+        files = next(os.walk(path))[2]  # Get all the files inside the list
+        for file in files:
+            if file.endswith(".json"):
+                self._download_metadata(file)
+
 
 def _ensure_trailing_slash(url: str) -> str:
     """Return url guaranteed to end in a slash"""
     return url if url.endswith("/") else f"{url}/"
+
+
+def _get_spec_version(
+    repository_versions: List[str],
+    spec_version: str,
+    supported_versions: List[str],
+) -> str:
+    """Returns the specification version to be used."""
+
+    repository_versions = [int(i) for i in repository_versions]
+    supported_versions = [int(i) for i in supported_versions]
+    spec_version = int(spec_version)
+
+    # The client determines the latest version available on the repository by looking
+    # for the directory with the largest version number.
+    latestrep_ver = max(repository_versions)
+
+    # If the latest version on the repository is lower than the previous specification
+    # version the client used from this repository, the client should report an error
+    # and terminate the update.
+    if spec_version:
+        if latestrep_ver < spec_version:
+            raise exceptions.DownloadError(
+                "The repository version is lower than the last used spec version"
+            )
+
+    # If the latest version on the client is found on the repository or vice versa, the
```

Comment:
Is this a typo? I can't find it in https://github.com/theupdateframework/taps/blob/840bbdaef85198e4b80e567b40b92fa592701bec/tap14.md and the phrase "latest version on the client" is confusing.

Reply:
Ahh, we decided to combine them, right.

I think you should copy all three relevant bullets:

> - If the latest version on the repository is equal to that of the client, it will use this directory to download metadata.
> - If the latest version pre-dates the client specification version, it may ...[use] a previous client version to download the metadata.
> If the latest version on the repository is higher than the client spec version, the client should ...perform the update with the directory that corresponds with the latest client specification version, if available. If no such directory exists, the client terminates the update.